modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -264,6 +264,27 @@ private boolean usesRetrofit2Library() {
                 }
             }
         }
+
+        // camelize path variables for Feign client
+        if ("feign".equals(getLibrary())) {
+            Map<String, Object> operations = (Map<String, Object>) objs.get("operations");
+            List<CodegenOperation> operationList = (List<CodegenOperation>) operations.get("operation");
+            for (CodegenOperation op : operationList) {
+                String path = new String(op.path);
+                String[] items = path.split("/", -1);
```

Comment:
Both the [2.0 spec](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/2.0.md#pathTemplating) and the [upcoming 3.0 spec](https://github.com/OAI/OpenAPI-Specification/blob/OpenAPI.next/versions/3.0.md#pathTemplating) are very vague about how path templating works:

> Path templating refers to the usage of curly braces ({}) to mark a section of a URL path as replaceable using path parameters.

I would argue this does not say it is restricted to `/`-separated segments.